### PR TITLE
 Task 4 get_article_by_id

### PR DESCRIPTION
**Pull request by WanwanShi**
add success test, invalid ID test and out of Range ID test into test file 
add '/api/articles/:article_id' into app.js 
add articles_controllers
add articles_models

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -57,4 +59,45 @@ describe("/api", ()=> {
...
+                    body: "some gifs",
+                    created_at: expect.any(String),
+                    article_img_url: "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+                    votes: 0,
+                })
+            )
+        })
+    })
+
+    test("GET 400: Responds with 400 err and msg 'Bad request:Invalid article_id type'",()=> {
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:86]: I would change this - you're not getting the second part of your err msg, so your description should reflect that :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -57,4 +59,45 @@ describe("/api", ()=> {
...
+        return request(app)
+        .get("/api/articles/Not_a_ID_type")
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:88]: Be careful with grammar here - 'a' should be 'an' when put before a word starting with a vowel :)

File: app.js [hunk 3]
```diff
@@ -2,16 +2,37 @@ const express = require("express");
 const app = express();
 const {getAllTopics} = require("./controllers/topics_controllers")
 const {getAllEndpoints} = require("./controllers/endpoints_controllers")
+const {getArticleById} = require("./controllers/articles_controllers")
 
 app.use(express.json());
```
Comment (nc-niamh-smith) [inline app.js:7]: Do you need this in your code yet? Remove this line until you need it (i.e. post / patch)

File: app.js [hunk 4]
```diff
@@ -2,16 +2,37 @@ const express = require("express");
...
+    }
+});
+app.use((err, req, res, next) => {
+    if (err.status) {
+        res.status(err.status).send({msg: err.msg});
+    } else {
+        next(err);
+    }
+});
+
```
Comment (nc-niamh-smith) [inline app.js:35]: Excellent error handling here! It might be nice to create a new file to put all these, so they don't take over your app.js :)

File: scratchPad.sql [hunk 5]
```diff
@@ -0,0 +1,3 @@
+\c nc_news_test
+
+SELECT * FROM articles;
```
Comment (nc-niamh-smith) [inline scratchPad.sql:3]: I would suggest adding this to your .gitignore! :)

**Review by nc-niamh-smith — COMMENTED** [review]
Excellent work here Wendy! I've left you a couple of comments re structure and a couple of oversights, but overall this is a really solid endpoint!
